arangod/Replication2/StateMachines/Document/DocumentFollowerState.cpp

```diff
@@ -106,4 +116,37 @@ auto DocumentFollowerState::applyEntries(
   });
 }
 
+/**
+ * @brief Using the underlying transaction handler to apply a local transaction,
+ * for this follower only.
+ */
+auto DocumentFollowerState::forceLocalTransaction(OperationType opType,
+                                                  velocypack::SharedSlice slice)
+    -> Result {
+  auto doc = DocumentLogEntry{std::string(shardId), opType, std::move(slice),
+                              TransactionId{0}.asFollowerTransactionId()};
+  if (auto applyRes = _transactionHandler->applyEntry(doc); applyRes.fail()) {
+    _transactionHandler->removeTransaction(TransactionId::createFollower());
+    return applyRes;
+  }
+  auto commit = DocumentLogEntry{std::string(shardId),
+                                 OperationType::kCommit,
+                                 {},
+                                 TransactionId{0}.asFollowerTransactionId()};
+  return _transactionHandler->applyEntry(commit);
+}
```

Comment:
The transaction id is used in multiple places, so we need to create it once and use it consistently:
```suggestion
  auto trxId = TransactionId::createFollower();
  auto doc = DocumentLogEntry{std::string(shardId), opType, std::move(slice), trxId};
  if (auto applyRes = _transactionHandler->applyEntry(doc); applyRes.fail()) {
    _transactionHandler->removeTransaction(trxId);
    return applyRes;
  }
  auto commit = DocumentLogEntry{std::string(shardId),
                                 OperationType::kCommit,
                                 {},
                                 trxId};
  return _transactionHandler->applyEntry(commit);
}
```